### PR TITLE
Shrey fix changes for team member list which showed already added member in the list and limit set to 10

### PR DESCRIPTION
**Pull request by Shreyj14**
# Description
Fix Teams showing members already there and limited view issue
i) Login as admin/Owner-> Other Links-> Teams-> In an existing team or create a new team -> Click on Add members icon
When the user searches for a team member to add, the search list should not show already added team members. It should only show active members not already on the team. 
ii) There should not be a limit on the number of  items that are displayed. Currently only 10 items are getting displayed

## Main changes explained:
Changed the filter in MemberAutoComplete.jsx file to add the existing member condition.
Also changed the dropdown style to fit the add team member modal, and removed the limit 10.
…

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. log as admin user
4. go to Other Links→ Teams→ Select Members for a team→ Add Team member
5. verify function that if a member is already in the team, he is not visible in the add list.
6. verify that add team member list doen not have a hard limit of 10, but is scrollable if there are more users.
7. The pop up message for addition is not a scope for this PR

## Screenshots or videos of changes:
Before Change

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/118795427/3b6b22f4-cc3d-4d23-b3a0-cb6bfbf45572

After Change

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/118795427/209447ef-b624-42ef-af61-45e6a6c3ed99

### CODE REVIEW

**Review by Bhagyashree444 — APPROVED** [review]
Both the functionalities work as expected. Great Job.

Screenshot (127)

PR #1352

**Review by vikrambadhan — APPROVED** [review]
Hi, i tested your PR again and it works as intended! Great job!


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/85417946/20100caf-3675-4463-83b4-9a2bf67f121b

**Comment by Vishala09** [comment]
I tested your PR!  Awesome work!!

It lists all the available users with the search criteria (more than 10)
It does not show the added user again. 
It displays the error `Please choose a valid user.` if I try to add the same user again or add an empty user.

Bu there is just one thing happening that's a little odd. When I search for an user and click the add button multiple times , it adds the user to the team for that number of times. Even after I reload , I can see multiple entries for the same user. (since I clicked the add button multiple times). I understand this is an edge case and doesn't happen a lot. And when I delete , all the entries of corresponding user gets deleted , since they are all tied to same ID in the back-end. 

I am leaving it to you to decide whether this needs to considered and fixed (since it's not a very common scenario)

Other than this , it works perfect. Great job!!

_Attaching a video of what I faced_
[Untitled_ Sep 29,2023 10_47 AM.webm](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/65274029/f5c16cd6-d6d4-4d8a-a59b-9db800579650)

**Comment by astrjc0326** [comment]
Hi @Shreyj14, I tested the PR, and it works as expected!
Test 1: For the member who is already in the team.
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/98345990/4907db2c-85fd-4183-ac92-64ec02b99d0a

Test 2: For the list to be scrollable.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/98345990/a6ea8f57-0cc5-4dd9-b005-6b28fd28dca2

Great Job!

Just some advice for the scrollable feature: I wonder if it would be more user-friendly if the scroll bar is inside the modal.
As follows:
Screen Shot 2023-09-29 at 11 07 24 AM

**Review by Vishala09 — CHANGES_REQUESTED** [review]
I have mentioned the details in my previous comment. I am requesting changes for the multiple click scenario. If you feel , it's unnecessary or unrelated to the PR, please let me know. I will approve the changes as it passes all the validations. Great Job!

**Comment by Shreyj14** [comment]
Hi @Vishala09 can you delete the cookies or check it in the incognito mode for this site and re-test it, because no one has faced this issue, and I am not able to replicate it. Thanks

**Comment by Shreyj14** [comment]
> Hi @Shreyj14, I tested the PR, and it works as expected! Test 1: For the member who is already in the team. https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/98345990/4907db2c-85fd-4183-ac92-64ec02b99d0a
> 
> Test 2: For the list to be scrollable.
> 
>  Screen.Recording.2023-09-29.at.10.52.48.AM.mov 
> Great Job!
> 
> Just some advice for the scrollable feature: I wonder if it would be more user-friendly if the scroll bar is inside the modal. As follows: Screen Shot 2023-09-29 at 11 07 24 AM

The scrollable feature for which you are suggesting is not in the scope of this PR, I only made the change for the dropdown list and not the team member list. 
Thank you

**Review by Shiwani99 — CHANGES_REQUESTED** [review]
Hi,
I tested your PR. Everything works as expected, but I encountered a problem while adding users to the team, so when I search for the users it is not showing the entire list of users that is I have an account called "shiwani test", and when I try to search for that user, I'm not able to find it in the list. I have attached a video below of how it works in my end and I have also checked the user management page to see if the user "shiwani test" exists which I have added in the video too. 


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/78906820/9a197119-3638-4daa-a754-109a76f8839d

**Review by JeffLi117 — APPROVED** [review]
Changes work as intended - when I search for a user, the list is not limited to only 10 results. Additionally, when I add a member to the team, they are immediately added to the team (and the change is seen without closing/opening again). 

image

Great job!

**Comment by Shreyj14** [comment]
> Hi, I tested your PR. Everything works as expected, but I encountered a problem while adding users to the team, so when I search for the users it is not showing the entire list of users that is I have an account called "shiwani test", and when I try to search for that user, I'm not able to find it in the list. I have attached a video below of how it works in my end and I have also checked the user management page to see if the user "shiwani test" exists which I have added in the video too.
> 
>  Screen.Recording.2023-09-29.at.4.32.30.PM.mov

Hi Shiwani,
From your recording what I understand is that, your users role are "Jerry Test Role' for shiwani test and shiwani test1. And the team match is probably just meant for Admin, Core Team, Manager, Volunteer, Mentor.  I dont know if that comes under my scope to change the role permissions.

**Comment by AnishPandita** [comment]
Hey tested the PR and the mentioned functionality seems to work well. The user is added correctly and once added does not show in the list. Also the results of search does not have a limit 10 and is scrollable. All seems correct according to the scope of this PR.
Screenshot 2023-09-29 at 2 12 35 PM
Screenshot 2023-09-29 at 2 11 07 PM
Screenshot 2023-09-29 at 2 10 49 PM

**Review by harshidad — CHANGES_REQUESTED** [review]
Hi . I tested PR. It works fine in both scenarios.
  verify function that if a member is already in the team, he is not visible in the add list.
  verify that the add team member list does not have a hard limit of 10, but is scrollable if there are more users.

But when the user already exists in the list . So I received the message "Please choose a valid user". After this message , I removed the user using the delete button. So the user was successfully removed from the list. but when I am trying to add the same user it still shows the message ""Please choose a valid user", but now the user is not in the list. Please see the attached video recording . 


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/109878936/fd41d382-dcf8-4838-877b-1d56c0a3caef

**Comment by Shreyj14** [comment]
> Hi . I tested PR. It works fine in both scenarios.   verify function that if a member is already in the team, he is not visible in the add list. verify that the add team member list does not have a hard limit of 10, but is scrollable if there are more users.
> 
> But when the user already exists in the list . So I received the message "Please choose a valid user". After this message , I removed the user using the delete button. So the user was successfully removed from the list. but when I am trying to add the same user it still shows the message ""Please choose a valid user", but now the user is not in the list. Please see the attached video recording . 
> 
>  PR.1352.mp4

Hi @harshidad the logic which I believe is that you need to select a name from the drop down list then only it is considered as a valid user. So when you delete the user, you might have to search your name again, and you will see that you are able to add it again. I haven't touched that logic of the code, so it needs to be a separate issue.

**Review by Shiwani99 — APPROVED** [review]
Hi,
I didn't know if certain users not being shown in dropdown is scope of this PR. Great work. Everything works fine.

**Review by harshidad — APPROVED** [review]
Hi . I tested PR. It works fine in both scenarios.
  verify function that if a member is already in the team, he is not visible in the add list.
  verify that the add team member list does not have a hard limit of 10, but is scrollable if there are more users.

But when the user already exists in the list . So I received the message "Please choose a valid user". After this message , I removed the user using the delete button. So the user was successfully removed from the list. but when I am trying to add the same user it still shows the message ""Please choose a valid user", but now the user is not in the list. Please see the attached video recording . 


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/109878936/fd41d382-dcf8-4838-877b-1d56c0a3caef

This mentioned issue needs to be covered separately. Hence approving .

**Review by DavidC0126 — APPROVED** [review]
Hi @Shreyj14 I tested your PR and it works as intended.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/74082492/e32cd41e-55ef-4300-825c-e8e7ab19686d

**Comment by emunkh404** [comment]
Hello @Shreyj14, I've given your pull request a test run, and it functions as expected.
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/145945063/315fc1fa-40af-4ab5-91cd-3dcd7eab83b1

**Review by TuanDinh140194 — APPROVED** [review]
I have test both function and it works as describe.
Screenshot 2023-09-30 125607

**Review by yangfan2001 — APPROVED** [review]
Hi, I have tested your PR and it worked as described.
image
image

**Review by Haoj1 — APPROVED** [review]
@Shreyj14 Good job! The member already in the team will not be in the list and there can be more than 10 member in a team.

the list

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/99279909/0983f39e-0b9d-42fa-8685-1fe5071dd6c6

The scrolling down

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/99279909/e3ea9e4a-7aff-478d-b10f-1a1c057e817d

**Review by Vishala09 — APPROVED** [review]

**Comment by Shreyj14** [comment]
@one-community it has 10 approvals.

**Comment by Shubhu0** [comment]
Hi @Shreyj14 Reviewer the PR and it is working as intended. Once added, a member that already exist does not show up when they are looked up on in the search field

image

And the member list is not just limited to 10 members and additional members can be added. Good job! 

image

**Review by zuhang2000 — COMMENTED** [review]
Hi I've tested your PR following your instructions! The functionality is working as expected.
First I test if it does not have a limit of 10 on scroll down menu. Then I also test about adding and deleting team members. I also test adding more than 10 team members. Working good!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/123051008/e88850b3-f05f-4fe3-967e-a4e2df238b4e

**Review by EdwinLauM — APPROVED** [review]
Hey @Shreyj14, tested your PR and works as expected! There is no hard limit for members in a team and names won't appear if they are already in the team

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/42288987/d726a766-5ae6-400e-a098-80bae69680a8

**Comment by beblicarl** [comment]
This functionality works as intended and the code is great!

[screencast-localhost_3000-2023.10.01-15_49_59.webm](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/68056104/c911220d-03cd-46c2-9913-5920332a7709)

**Review by beblicarl — APPROVED** [review]

**Comment by one-community** [comment]
Thank you everyone, merging!

**Pull request merged**
